### PR TITLE
BACK-2813 Allow collecting additional patient details when accepting …

### PR DESCRIPTION
**Pull request by toddkazakov**
…an invite

### CODE REVIEW

**Review by ewollesen — DISMISSED** [review]
I'm approving this, as there are no show stoppers. However, this will have some conflicts with #134, which is just waiting for approval to merge.

In addition, 134 brings in some improved error handling, but you can't use it unless 134 is already merged of course.

I suppose if this merges first, I could quickly update it after 134 merges, so no biggie either way.

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ dist: focal
 language: go
 
 go:
-  - 1.19
```
[the remark this answers is not in the document]
Comment (toddkazakov) [reply .travis.yml:8]: The build was failing so I had to bump it up. This is a high-priority bugfix so we will see who gets to merge PRs first.

File: api/patientInvites.go [hunk 2]
```diff
@@ -96,7 +98,36 @@ func (a *Api) AcceptPatientInvite(res http.ResponseWriter, req *http.Request, va
...
 
-		patient, err := a.createClinicPatient(ctx, *conf)
+		accept := models.AcceptPatientInvite{}
+		if err := json.NewDecoder(req.Body).Decode(&accept); err != nil {
+			a.logger.Errorw("error decoding accept patient invite body", zap.Error(err))
+			a.sendModelAsResWithStatus(
+				res,
+				&status.StatusError{Status: status.NewStatus(http.StatusInternalServerError, STATUS_ERR_CREATING_PATIENT)},
+				http.StatusInternalServerError,
+			)
```
[the remark this answers is not in the document]
Comment (toddkazakov) [reply api/patientInvites.go]: Thanks, if you end up merging your PR first I'll fix this.

File: models/confirmation.go [hunk 3]
```diff
@@ -50,6 +50,13 @@ type (
 		RequiredIdp string `json:"requiredIdp,omitempty"`
 	}
 
+	AcceptPatientInvite struct {
+		Mrn       string   `json:"mrn"`
```
[the remark this answers is not in the document]
Comment (toddkazakov) [reply models/confirmation.go]: Fixed

**Comment by ewollesen** [comment]
Please don't hate me, but #134 merged and there are lots of conflicts.

File: api/patientInvites.go [hunk 4]
```diff
@@ -91,7 +94,34 @@ func (a *Api) AcceptPatientInvite(res http.ResponseWriter, req *http.Request, va
 			return
 		}
 
-		patient, err := a.createClinicPatient(ctx, *conf)
+		accept := models.AcceptPatientInvite{}
+		if req.ContentLength > 0 {
```
Comment (ewollesen) [inline api/patientInvites.go:98]: Under what circumstances is it OK for the request's body to be empty?

The body was previously required, but now is not. Is it simply optional now?

File: api/patientInvites.go [hunk 5]
```diff
@@ -91,7 +94,34 @@ func (a *Api) AcceptPatientInvite(res http.ResponseWriter, req *http.Request, va
...
+			if err := json.NewDecoder(req.Body).Decode(&accept); err != nil {
+				a.sendError(ctx, res, http.StatusInternalServerError, STATUS_ERR_CREATING_PATIENT,
+					fmt.Errorf("error decoding accept patient invite body: %w", err),
```
Comment (ewollesen) [inline api/patientInvites.go:101]:
```suggestion
					fmt.Errorf("decoding accept patient invite body: %w", err),
```
nit: it's already an error value, no need to repeat it

**Review by ewollesen — APPROVED** [review]
I wish there'd been time to add tests for the new functionality.

**Pull request merged**